src/XMakeBuildEngine/Evaluation/LazyItemEvaluator.LazyItemOperation.cs

```diff
@@ -59,9 +59,24 @@ IList<I> GetReferencedItems(string itemType, ImmutableHashSet<string> globsToIgn
                 }
             }
 
-            public abstract void Apply(ImmutableList<ItemData>.Builder listBuilder, ImmutableHashSet<string> globsToIgnore);
+            public virtual void Apply(ImmutableList<ItemData>.Builder listBuilder, ImmutableHashSet<string> globsToIgnore)
+            {
+                var items = SelectItems(listBuilder, globsToIgnore).ToList();
+                MutateItems(items);
```

Comment:
@cdmihai OK, so we aren't currently caching the items in a `LazyItemList`.  So every time `GetItems` is called, everything is reevaluated from scratch, and a new set of items is created.

We probably want to cache the list of items we return from `LazyItemList.GetItems`.  If we do, then with the current code the metadata update of `i2` would probably inadvertently apply to the items from the previous version of `i2` that were "copied" into `i`.